### PR TITLE
Kafka Scaler: added support for PKCS #8 private key decryption

### PR DESCRIPTION
**Pull request by tobiaskrause**
Signed-off-by: Tobias Krause <tobias.krause@otto.de>

Enhanced Kafka TLS configuration to accept a passphrase for a provided PKCS #8 private key. 

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes https://github.com/kedacore/keda/issues/3449

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 
 ## Unreleased
 
+- **Kafka Scaler:** Support of passphrase encrypted PKCS #\8 private key [3413](https://github.com/kedacore/keda/pull/3413)
```
Comment (zroubalik) [inline CHANGELOG.md:37]: Please move this to `Improvements Section below`, sort alphabetically and please refer an issue (not PR in the Changelog). If there's not an issue covering this feature request, please create one.

**Review by zroubalik — COMMENTED** [review]
Looking good, could you please fix the Changelog and open docs PR to update documentation for upcoming (2.8) version?
https://github.com/kedacore/keda-docs/pulls

**Comment by tobiaskrause** [comment]
PR for doc: https://github.com/kedacore/keda-docs/pull/844

**Review by zroubalik — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

**Pull request merged**